ui/src/reusable_ui/components/button/Button.tsx

```diff
@@ -0,0 +1,96 @@
+// Libraries
+import React, {Component} from 'react'
+import classnames from 'classnames'
+
+// Types
+import {
+  ComponentColor,
+  ComponentSize,
+  ButtonShape,
+  IconFont,
+} from 'src/reusable_ui/types'
+import {RemoteDataState} from 'src/types/flux'
```

Comment:
you might consider making a new enum type that is more component-oriented, rather than AJAX-oriented, which could be a union of RemoteDataState and any other states you might want to contemplate in these components